velox/exec/tests/MultiFragmentTest.cpp

```diff
@@ -1016,6 +1016,127 @@ TEST_F(MultiFragmentTest, cancelledExchange) {
   EXPECT_EQ(1, exchangeTask.use_count());
 }
 
+class TestCustomExchangeNode : public core::PlanNode {
+ public:
+  TestCustomExchangeNode(
+      const core::PlanNodeId& id,
+      const RowTypePtr type,
+      bool* used)
+      : PlanNode(id), outputType_(type), used_(used) {}
+
+  const RowTypePtr& outputType() const override {
+    return outputType_;
+  }
+
+  bool* used() const {
```

Comment:
This API looks a bit strange. I'm running out of time now. Will take a closer look later.

Reply:
It's mainly an easy way for the purpose of testing to see if this operator is used.